File: 15-01-subclass-view/java-15-01-subclass-view/app/src/main/java/com/example/subclassview/MainActivity.java
package com.example.subclassview;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView mTextView;
    private EditText mEditText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mTextView = findViewById(R.id.text);
        mEditText = findViewById(R.id.editText);

        Button button = findViewById(R.id.button);

        button.setOnClickListener(v -> mTextView.setText(mEditText.getText().toString()));
    }
}

File: 15-01-subclass-view/kotlin-14-01-subclass-view/app/src/main/java/com/example/subclassview/MainActivity.kt
package com.example.subclassview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    private var mTextView: TextView? = null
    private var mEditText: EditText? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mTextView = findViewById(R.id.text)
        mEditText = findViewById(R.id.editText)

        val button = findViewById<Button>(R.id.button)

        button.setOnClickListener { v -> mTextView!!.text = mEditText!!.text.toString() }
    }
}
